### PR TITLE
dynamic-layers: move Firefox to rust-layer

### PR DESCRIPTION
**Pull request by cpboyd**
Per https://github.com/OSSystems/meta-browser/issues/237#issuecomment-495435011

As the title states, this is moving the `firefox` recipe to `dynamic-layers/rust-layer/`.

I opted not to move the other Firefox-related recipes since they don't implicitly depend on meta-rust.

Granted, **firefox-l10n.inc** and **firefox-addon.inc** contain this commented line:
`# RDEPENDS_${PN} = "firefox"`

If it's uncommented, they'll need to be moved to the **rust-layer** folder

### CODE REVIEW

**Comment by rakuco** [comment]
cc @otavio @kraj, I'm not familiar with dynamic layers and I guess you guys are the best to comment on this

**Comment by kraj** [comment]
@cpboyd dont they depend upon firefox ? and whats the point of having them without firefox ?

**Comment by ashie** [comment]
FYI I'm now working on Firefox 68: https://github.com/webdino/meta-browser/tree/firefox-68-wip
From this version Firefox will depend on both meta-rust & meta-clang because recent firefox can't build with GCC (and I don't want to spent my time to address them).

I'm not familiar with dynamic-layers, so I'm now wondering how to solve such case with this directory structure?

**Comment by cpboyd** [comment]
@kraj As I mentioned, the BitBake recipes don't currently seem to... (the `RDEPENDS` is commented out)
Perhaps for organization purposes, the other Firefox-related recipes should be moved to the rust-layer too.

@ashie Interesting point that's be related to #237.
I suppose if that's the case, then meta-clang should be a hard requirement for meta-browser (as both Chromium and Firefox require it).

**Comment by otavio** [comment]
@cpboyd I'd prefer to have them all under the same namespace. Also, it seems that the recipes need to be fixed to rdepends on firefox. 

Could you add extra commits to address it?

**Comment by cpboyd** [comment]
@otavio Sure, no problem.

However, with regards to the `RDEPENDS`, this commit https://github.com/OSSystems/meta-browser/commit/176d9edd1148064e94ff3fe7f957a057d3acd614 implies that the dependency on `firefox` was dropped to keep `inherit allarch`.

With that in mind, should I still revert the changes from that commit?

**Comment by otavio** [comment]
We can add the ABI whitelist for it, so it does not get rebuild when firefox architecture changes.

**Comment by cpboyd** [comment]
@otavio I'm not familiar with ABI whitelisting.  Is there a field to be added to the BitBake recipes dependent upon Firefox?

**Comment by otavio** [comment]
> @otavio I'm not familiar with ABI whitelisting. Is there a field to be added to the BitBake recipes dependent upon Firefox?

Like this:

https://patchwork.openembedded.org/patch/144057/

**Comment by cpboyd** [comment]
I've successfully rebuilt my own project, but I'm facing this issue when I try to build Firefox: https://github.com/meta-rust/meta-rust/issues/240

File: conf/layer.conf [hunk 1]
```diff
@@ -9,6 +9,102 @@ BBFILES_DYNAMIC += " \
 	rust-layer:${LAYERDIR}/dynamic-layers/rust-layer/*/*/*.bbappend \
 "
 
+SIGGEN_EXCLUDERECIPES_ABISAFE_rust-layer += " \
+   firefox-addon-webconverger \
+   firefox-l10n-ach \
```
Comment (cpboyd) [inline conf/layer.conf]: A potential concern here is that if more Firefox localizations are added, they'll need to be added to this list.

Comment (otavio) [reply conf/layer.conf]: Agreed but it is still doable and allow for correct dependency chain.

**Pull request merged**